### DISCUSSION TITLE
Same XML recipe Import for all Tenants

### DISCUSSION THREAD

**Discussion by msquare98** [OrchardCMS/OrchardCore#10133] [Q&A]
I am working on ImportExport module of orchard CMS. I am new to Orchard CMS and trying to import one recipe for all Tenants including the default. It may be a very basic question, sorry if it is, i am unable to execute the recipe using the name of specific tenant. Is it possible? or am i am doing it wrong? I want to use ITenantService and get all tenants and then loop over all tenants and execute the same recipe to each tenant. I can't find any execute method which includes tenant name. All i have in the modules is importing or executing to defualt recipe. Is there a way to execute recipe using the tenant name or ID? Sorry if it is a noob question, i am looking for clarification on this one. Thanks in advance

**Comment by Skrypt** [comment]
There is an autosetup feature to populate multiple tenants at once.
https://docs.orchardcore.net/en/dev/docs/reference/modules/AutoSetup/

**Reply by msquare98** [reply]
Thanks for the answer, but this link talks about setting up tenants, what i want is to import the recipe files to exisiting tenants along with deafult tenant

**Reply by deanmarcussen** [reply]
I think @jptissot added a feature like this to his repository https://github.com/statcan/statcan.orchardcore/ 

I suggested it could be potentially good as a feature where tenants could become a deployment target

Equally you can also setup tenants as a remote deployment target already and send deployment plans directly to them

**Reply by msquare98** [reply]
Thanks, i will check the repository and hopefully, i can find a solution

**Comment by jtkech** [comment]
@destro98 

You can't execute something for a given tenant / shell in the context of another tenant, we have `IShellHost` / `ShellScope` methods / helpers to get a `ShellScope` under a `ShellContext` container related to a given tenant / shell, and then an `UsingAsync(scope => )` helper to execute something in the context of this `ShellScope`.

As @Skrypt said you can look up at the autosetup module, from memory the openid module also uses some of these helpers in its configurations. Finally, a tenant being like a site, as @deanmarcussen said you could use existing deployment features.

**Reply by Skrypt** [reply]
What we will eventually need is some sort of Tenant Management UI where we can do these kind of batch operations.

**Reply by msquare98** [reply]
Thanks for the answers @jtkech @Skrypt i have to understand better about shell to do this, i will read more into it. Thanks again
